airflow/providers/alibaba/cloud/hooks/analyticdb_spark.py

```diff
@@ -0,0 +1,377 @@
+    FAILED = "FAILED"
+    KILLING = "KILLING"
+    KILLED = "KILLED"
+    SUCCEEDING = "SUCCEEDING"
+    COMPLETED = "COMPLETED"
+    FATAL = "FATAL"
+    UNKNOWN = "UNKNOWN"
+
+
+class AnalyticDBSparkHook(BaseHook, LoggingMixin):
+    """
+    Hook for AnalyticDB MySQL Spark through the REST API.
+
+    :param adb_spark_conn_id: The Airflow connection used for AnalyticDB MySQL Spark credentials.
+    :param region: AnalyticDB MySQL region you want to submit spark application.
+    """
+
+    TERMINAL_STATES = {
+        AppState.COMPLETED,
+        AppState.FAILED,
+        AppState.FATAL,
+        AppState.KILLED
+    }
+
+    conn_name_attr = "alibabacloud_conn_id"
+    default_conn_name = "adb_spark_default"
+    conn_type = "adb_spark"
+    hook_name = "AnalyticDB Spark"
+
+    def __init__(
+        self,
+        adb_spark_conn_id: str = "adb_spark_default",
+        region: str | None = None,
+        *args,
+        **kwargs
+    ) -> None:
+        self.adb_spark_conn_id = adb_spark_conn_id
+        self.adb_spark_conn = self.get_connection(adb_spark_conn_id)
+        self.region = self.get_default_region() if region is None else region
+        super().__init__(*args, **kwargs)
+
+    def submit_spark_app(
+        self,
+        cluster_id: str,
+        rg_name: str,
+        *args: Any,
+        **kwargs: Any
+    ) -> SubmitSparkAppResponse:
+        """
+        Perform request to submit spark application
+
+        :param cluster_id: The cluster ID of AnalyticDB MySQL 3.0 Data Lakehouse.
+        :param rg_name: The name of resource group in AnalyticDB MySQL 3.0 Data Lakehouse cluster.
+        """
+        self.log.info("Submitting application")
+        try:
+            request = SubmitSparkAppRequest(
+                dbcluster_id=cluster_id,
+                resource_group_name=rg_name,
+                data=json.dumps(self.build_submit_app_data(*args, **kwargs)),
+                app_type="BATCH"
+            )
+            return self.get_adb_spark_client().submit_spark_app(request)
+        except Exception as e:
+            self.log.error(e)
+            raise AirflowException("Errors when submit spark application")
+
+    def submit_spark_sql(
+        self,
+        cluster_id: str,
+        rg_name: str,
+        *args: Any,
+        **kwargs: Any
+    ) -> SubmitSparkAppResponse:
+        """
+        Perform request to submit spark sql
+
+        :param cluster_id: The cluster ID of AnalyticDB MySQL 3.0 Data Lakehouse.
+        :param rg_name: The name of resource group in AnalyticDB MySQL 3.0 Data Lakehouse cluster.
+        """
+        self.log.info("Submitting Spark SQL")
+        try:
+            request = SubmitSparkAppRequest(
+                dbcluster_id=cluster_id,
+                resource_group_name=rg_name,
+                data=self.build_submit_sql_data(*args, **kwargs),
+                app_type="SQL"
+            )
+            return self.get_adb_spark_client().submit_spark_app(request)
```

Comment:
```suggestion
            request = SubmitSparkAppRequest(
                dbcluster_id=cluster_id,
                resource_group_name=rg_name,
                data=self.build_submit_sql_data(*args, **kwargs),
                app_type="SQL"
            )
        try:
            return self.get_adb_spark_client().submit_spark_app(request)
```

Reply:
I'm assuming you only expect `self.get_adb_spark_client().submit_spark_app(request)` to raise an exception and not request the creation part?